demos/top-app-bar.html

```diff
@@ -203,7 +208,17 @@ <h3>Demo Controls</h3>
           if (!this.checked) {
             appBarEl.classList.remove('mdc-top-app-bar--short-collapsed');
             appBarEl.classList.remove('mdc-top-app-bar--short-has-action-item');
+            alwaysClosedCheckbox.checked = false;
           }
+
+          alwaysClosedCheckbox.disabled = !this.checked;
```

Comment:
I'd suggest adding a disabled label style like in our text field demo page, to make the disabled state more obvious.